firmware/hackrf_usb/usb_api_transceiver.c

```diff
@@ -262,29 +274,34 @@ void set_transceiver_mode(const transceiver_mode_t new_transceiver_mode) {
 		led_off(LED3);
 		led_on(LED2);
 		rf_path_set_direction(&rf_path, RF_PATH_DIRECTION_RX);
-		usb_bulk_buffer_tx = false;
+		set_m0_mode(M0_MODE_RX);
```

Comment:
It looks like this (and other mode changes) will result in the correct behavior only if the host has first asked us to set the mode to IDLE, but nothing enforces that. I feel like this should return an error to the host if the current mode does not support a direct transition to the requested mode. However, it might be better to accept whatever the host asks and forward the request to the M0 (with some changes to the M0 code to handle mode change requests (rather than having the M4 clobber the M0's mode directly).

Reply:
Fixed in cb3cfe8